Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/BaseInteractiveElement.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License
+
+using Microsoft.MixedReality.Toolkit.Input;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// Base class for an Interaction Element.  Contains state management methods, event management and the state setting logic for
+    /// Core Interaction States.
```

Comment:
Core Interaction States or all states (core + user-defined)?

Reply:
BaseInteractiveElement supports adding core interaction states and creating user defined states.  But only the state setting logic for a core state is in BaseInteractiveElement.  If a user were to create a state, the state setting logic would not be added to BaseInteractiveElement, it would be in a separate file.